packages/ui/react-ui-editor/src/components/TextEditor/extensions/hyperlinkDecoration.ts

```diff
@@ -0,0 +1,99 @@
+// https://discuss.codemirror.net/t/link-widget-is-clicked-only-if-editor-is-out-of-focus/7433
+
+// TODO(burdon): Adapt to make all markdown editable?
+
+// TODO(burdon): Research overlays.
+//  https://stackoverflow.com/questions/26933344/codemirror-detect-and-create-links-inside-editor
+//  https://codemirror.net/5/doc/manual.html#addOverlay
+//  https://codemirror.net/5/demo/simplemode.html
+//  https://codemirror.net/5/demo/search.html (Search dialog).
+
+class Link extends WidgetType {
+  constructor(private readonly _anchor: number, private readonly _text: string, private readonly _url?: string) {
+    super();
+  }
+
+  override eq(other: any) {
+    return this._text === other.text && this._url === other.url;
+  }
+
+  toDOM(view: EditorView) {
+    const link = document.createElement('a');
+    link.setAttribute('class', 'cm-link');
```

Comment:
`cm-` is CodeMirror’s namespace, which we may like to avoid, though I don’t necessarily have an opinion at the moment.

We can define a classname in theme.css that aligns this with our Link component. In fact, that may be the most sustainable way to go.

Reply:
given that this in inline with the CM content i was deliberately extending their class system (and i think this was suggested by marijn the instigator in his snippet.) but i do think we need to reconcile our CM theme with our styling in general (e.g., colors, etc.)